tiny_dnn/util/serialization_functions.h

```diff
@@ -391,41 +391,6 @@ struct specialize<Archive,
                   tiny_dnn::slice_layer,
                   cereal::specialization::non_member_serialize> {};
 
-template <class Archive>
-struct specialize<Archive,
-                  tiny_dnn::sigmoid_layer,
-                  cereal::specialization::non_member_serialize> {};
-
```

Comment:
@karandesai-96 why do you remove this? I propose to adopt a templated routine instead
```cpp
template <class Archive, typename L>
  void serialize(Archive& ar, L& layer) {
    serialization_buddy::serialize(ar, layer);
}
```